docker/templates/Dockerfile.erb

```diff
@@ -1,148 +1,147 @@
-<%   locale = 'en_US.UTF-8' -%>
-<% end -%>
 <% if image_flavor == 'ironbank' -%>
-ARG BASE_REGISTRY=registry1.dsop.io
+ARG BASE_REGISTRY=registry1.dso.mil
 ARG BASE_IMAGE=ironbank/redhat/ubi/ubi9
-ARG BASE_TAG=9.2
+ARG BASE_TAG=9.3
 ARG LOGSTASH_VERSION=<%= elastic_version %>
-ARG GOLANG_VERSION=1.17.8
+ARG GOLANG_VERSION=1.21.8
 
-FROM ${BASE_REGISTRY}/${BASE_IMAGE}:${BASE_TAG} AS env2yaml
+# stage 1: build env2yaml
+FROM ${BASE_REGISTRY}/google/golang/ubi9/golang-1.21:${GOLANG_VERSION} AS env2yaml
 
-ARG GOLANG_VERSION
+ENV GOPATH=/go
 
-# install golang
-RUN yum update -y && yum install -y git
-COPY go${GOLANG_VERSION}.linux-amd64.tar.gz /opt/go.tar.gz
-RUN tar -C /usr/local -xzf /opt/go.tar.gz
-ENV PATH=$PATH:/usr/local/go/bin
+COPY scripts/go /go
 
-# compile the env2yaml tool
-COPY v2.3.0.tar.gz /opt/env2yaml.tar.gz
-COPY scripts/go /usr/local/src/go
-WORKDIR /usr/local/src/go/src/env2yaml
-RUN mkdir -p vendor/gopkg.in
-RUN tar -zxf /opt/env2yaml.tar.gz -C vendor/gopkg.in
-RUN mv vendor/gopkg.in/yaml-2.3.0 vendor/gopkg.in/yaml.v2
-ENV GOPATH=/usr/local/src/go
-RUN go build -mod vendor
+USER root
 
-# stage 1: unpack logstash
-FROM ${BASE_REGISTRY}/${BASE_IMAGE}:${BASE_TAG} AS builder
+RUN dnf-3 -y upgrade && dnf-3 install -y git && \
+  cd /go/src/env2yaml && \
+  go build
+
+# Final stage
+FROM ${BASE_REGISTRY}/${BASE_IMAGE}:${BASE_TAG}
 
 ARG LOGSTASH_VERSION
 
-WORKDIR /usr/share/
-COPY logstash-${LOGSTASH_VERSION}-linux-x86_64.tar.gz /opt/logstash.tar.gz
+ENV ELASTIC_CONTAINER true
+ENV PATH=/usr/share/logstash/bin:$PATH
 
-RUN tar zxf /opt/logstash.tar.gz && \
-    mv /usr/share/logstash-${LOGSTASH_VERSION} /usr/share/logstash
+WORKDIR /usr/share
+
+COPY --from=env2yaml /go/src/env2yaml/env2yaml /usr/local/bin/env2yaml
+COPY scripts/config/* config/
+COPY scripts/pipeline/default.conf pipeline/logstash.conf
+COPY scripts/bin/docker-entrypoint /usr/local/bin/
+COPY logstash-${LOGSTASH_VERSION}-linux-x86_64.tar.gz /tmp/logstash.tar.gz
+
+RUN dnf -y upgrade && \
+  dnf install -y procps findutils tar gzip which shadow-utils && \
+  dnf clean all && \
+  groupadd --gid 1000 logstash && \
+  adduser --uid 1000 --gid 1000 --home-dir /usr/share/logstash --no-create-home logstash && \
+  tar -zxf /tmp/logstash.tar.gz -C /usr/share/ && \
+  mv /usr/share/logstash-${LOGSTASH_VERSION} /usr/share/logstash && \
+  chown -R 1000:0 /usr/share/logstash && \
+  chown --recursive logstash:logstash /usr/share/logstash/ && \
+  chown -R logstash:root /usr/share/logstash config/ pipeline/ && \
+  chmod -R g=u /usr/share/logstash && \
+  mv config/* /usr/share/logstash/config && \
+  mv pipeline /usr/share/logstash/pipeline && \
+  mkdir /licenses/ && \
+  mv /usr/share/logstash/NOTICE.TXT /licenses/NOTICE.TXT && \
+  mv /usr/share/logstash/LICENSE.txt /licenses/LICENSE.txt && \
+  ln -s /usr/share/logstash /opt/logstash && \
+  chmod 0755 /usr/local/bin/docker-entrypoint && \
+  rmdir config && \
+  rm /tmp/logstash.tar.gz
```

Comment:
Can we add a comment to state that this closes the `<% if image_flavor == 'ironbank' -%>`